lib/gollum-lib/filter/toc.rb

```diff
@@ -24,15 +25,38 @@ def process(data)
         add_anchor_to_header header, anchor_name
         add_entry_to_toc     header, anchor_name
       end
+      if not @toc_doc.nil?
+        toc_str = @toc_doc.to_xml(@markup.to_xml_opts)
+      end
 
-      @toc  = @toc.to_xml(@markup.to_xml_opts) if @toc != nil
       data  = @doc.to_xml(@markup.to_xml_opts)
-      
     end
- 
-    @markup.toc = @toc
-    data.gsub("[[_TOC_]]") do
-      @toc.nil? ? '' : @toc
+
+    @markup.toc = toc_str
+
+    data.gsub(/\[\[_TOC_(.*?)\]\]/) do
+      levels = nil
+
+      opts = Regexp.last_match[1].split('|')[1..-1] || []
+      opts.each do |attr|
```

Comment:
Seems like this is quite an expensive operation if all we want is the value of levels. I think a regexp would be much faster, or not?